frontend/src/pages/RunList.tsx

```diff
@@ -376,16 +376,42 @@ class RunList extends React.PureComponent<RunListProps, RunListState> {
    * DisplayRun will show '-'.
    */
   private async _getAndSetExperimentNames(displayRun: DisplayRun): Promise<void> {
-    const experimentId = RunUtils.getFirstExperimentReferenceId(displayRun.run);
-    if (experimentId) {
-      try {
-        const experiment = await Apis.experimentServiceApi.getExperiment(experimentId);
-        displayRun.experiment = { displayName: experiment.name || '', id: experimentId };
-      } catch (err) {
-        // This could be an API exception, or a JSON parse exception.
-        displayRun.error = 'Failed to get associated experiment: ' + await errorToMessage(err);
+    const experimentInfo = this._getFirstExperimentInfoInReferences(displayRun.run.resource_references);
+    if (experimentInfo) {
+      displayRun.experiment = experimentInfo;
+    } else {
+      const experimentId = RunUtils.getFirstExperimentReferenceId(displayRun.run);
+      if (experimentId) {
+        try {
+          const experiment = await Apis.experimentServiceApi.getExperiment(experimentId);
+          displayRun.experiment = { displayName: experiment.name || '', id: experimentId };
+        } catch (err) {
+          displayRun.error = 'Failed to get associated experiment: ' + await errorToMessage(err);
+        }
+      }
+    }
+  }
+
+  /**
+   * Checks all provided ApiResourceReference objects to see if they provide
+   * valid data to populate an ExperimentInfo object (name, id, and the type is
+   * ApiResourceType.EXPERIMENT). If conditions are met, the
+   * ApiResourceReference is returned as an ExperimentInfo object, otherwise,
+   * undefined is returned if no valid data is found. 
+   */
+  private _getFirstExperimentInfoInReferences(resourceReferences: ApiResourceReference[] = []): ExperimentInfo | undefined {
+    for (const resourceReference of resourceReferences) {
```

Comment:
nitpick: this could be simpler by reusing existing util functions like [this](https://github.com/kubeflow/pipelines/pull/1837/files#diff-79233ec80e41748d7203b325c5a8a0ddR71)

`RunUtils.getAllExperimentReferences` can deal with finding all experiment references for you.